releasenotes/notes/improve-qsd-zxz-block-decomposition-73a1a9e155ccc809.yaml

```
@@ -0,0 +1,16 @@
+---
+upgrade_synthesis:
+  - |
+    Improve the :func:`.qs_decomposition` function which was originally based on
+    Shende et. al. (https://arxiv.org/abs/quant-ph/0406176).
+    The new synthesis method is based on Krol and Al-Ars (https://arxiv.org/abs/2403.13692),
+    and reduce the total number of :class:`.CXGate`\ s of a general n-qubit unitary by
+    (4 ** (n − 2) − 1) / 3 :class:`.CXGate`\ s.
+    With the improved decomposition, a general 3-qubit unitary can be decomposed using 19
+    :class:`.CXGate`\ s (rather than 20).
+  - |
```

The first note is a feature note, the second is an upgrade note:

```suggestion
features_synthesis:
  - |
    Improve the :func:`.qs_decomposition` function which was originally based on
    Shende et. al. (https://arxiv.org/abs/quant-ph/0406176).
    The new synthesis method is based on Krol and Al-Ars (https://arxiv.org/abs/2403.13692),
    and reduce the total number of :class:`.CXGate`\ s of a general n-qubit unitary by
    (4 ** (n − 2) − 1) / 3 :class:`.CXGate`\ s.
    With the improved decomposition, a general 3-qubit unitary can be decomposed using 19
    :class:`.CXGate`\ s (rather than 20).
upgrade_synthesis:
  - |
```